src/mpi/group/group_impl.c

```diff
@@ -633,32 +635,71 @@ int MPIR_Group_from_session_pset_impl(MPIR_Session * session_ptr, const char *ps
     int mpi_errno = MPI_SUCCESS;
     MPIR_Group *group_ptr;
 
-    if (MPL_stricmp(pset_name, "mpi://WORLD") == 0) {
-        mpi_errno = MPIR_Group_create(MPIR_Process.size, &group_ptr);
+    if (MPL_stricmp(pset_name, MPIR_SESSION_WORLD_PSET_NAME) == 0) {
+        mpi_errno = MPIR_Group_create(session_ptr->size, &group_ptr);
         MPIR_ERR_CHECK(mpi_errno);
 
-        group_ptr->size = MPIR_Process.size;
-        group_ptr->rank = MPIR_Process.rank;
+        group_ptr->size = session_ptr->size;
+        group_ptr->rank = session_ptr->rank;
         group_ptr->is_local_dense_monotonic = TRUE;
         for (int i = 0; i < group_ptr->size; i++) {
             group_ptr->lrank_to_lpid[i].lpid = i;
             group_ptr->lrank_to_lpid[i].next_lpid = i + 1;
         }
         group_ptr->lrank_to_lpid[group_ptr->size - 1].next_lpid = -1;
         group_ptr->idx_of_first_lpid = 0;
-    } else if (MPL_stricmp(pset_name, "mpi://SELF") == 0) {
+
+    } else if (MPL_stricmp(pset_name, MPIR_SESSION_SELF_PSET_NAME) == 0) {
         mpi_errno = MPIR_Group_create(1, &group_ptr);
         MPIR_ERR_CHECK(mpi_errno);
 
         group_ptr->size = 1;
         group_ptr->rank = 0;
         group_ptr->is_local_dense_monotonic = TRUE;
-        group_ptr->lrank_to_lpid[0].lpid = MPIR_Process.rank;
+        group_ptr->lrank_to_lpid[0].lpid = session_ptr->rank;
         group_ptr->lrank_to_lpid[0].next_lpid = -1;
         group_ptr->idx_of_first_lpid = 0;
     } else {
-        /* TODO: Implement pset struct, locate pset struct ptr */
-        MPIR_ERR_SETANDSTMT(mpi_errno, MPI_ERR_ARG, goto fn_fail, "**psetinvalidname");
+        MPIR_Pset *pset = NULL;
+        mpi_errno = MPIR_Pset_by_name(MPIR_Process.pm_pset_array, pset_name, &pset);
+
+        if (mpi_errno == MPI_ERR_OTHER || pset == NULL) {
+            *new_group_ptr = NULL;
+            MPIR_ERR_SETANDSTMT(mpi_errno, MPI_ERR_ARG, goto fn_fail, "**psetinvalidname");
+        }
+
+        /* Remark: We do not check the pset for validity here. Creating a Process Group
+         * from an invalid pset should be possible as a local operation. However, creating
+         * an MPI communicator from a Process Group that is based on an invalid pset
+         * should raise an error. */
+
+        mpi_errno = MPIR_Group_create(pset->size, &group_ptr);
+        MPIR_ERR_CHECK(mpi_errno);
+
+        /* Remark: The following logic assumes that pset->members is sorted in ascending order of ranks */
+        for (int i = 0; i < pset->size; i++) {
+            /*group rank is index of global rank in sorted array */
+            if (pset->members[i] == session_ptr->rank) {
```

Comment:
We'll need a global PMIx Process Id -> rank translation here.

Reply:
Indeed, this is a todo for the future, when we add support for dynamic resources/ malleability. Added a [comment](https://github.com/pmodels/mpich/pull/6430/files#diff-a7fe0a05ef65ff6d4a4ad561c5c04946f1c9cc49ca0c4821823bb36c2ea4f4daR407) that explains our plans for this translation.